crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -65,6 +65,667 @@ impl Circle {
     }
 }
 
+const HALF_PI: f32 = PI / 2.0;
+
+/// A primitive representing an arc between two points on a circle.
+///
+/// An arc has no area.
+/// If you want to include the portion of a circle's area swept out by the arc,
+/// use [`CircularSector`].
+/// If you want to include only the space inside the convex hull of the arc,
+/// use [`CircularSegment`].
+///
+/// The arc is drawn starting from [`Vec2::Y`], extending by `half_angle` radians on
+/// either side. The center of the circle is the origin [`Vec2::ZERO`]. Note that this
+/// means that the origin may not be within the `Arc2d`'s convex hull.
+///
+/// **Warning:** Arcs with negative angle or radius, or with angle greater than an entire circle, are not officially supported.
```

Comment:
Hmm. Why do you think we should allow these be constructed, rather than enforcing this via the API?

Reply:
I don't feel strongly, but it's consistent with the other primitives to expose the fields and thereby let you construct invalid ones.